ProcessLib/HT/StaggeredHTFEM-impl.h

```diff
@@ -0,0 +1,286 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *  \file   StaggeredHTFEM-impl.h
+ *  Created on October 13, 2017, 3:52 PM
+ */
+
+#pragma once
+
+#include "StaggeredHTFEM.h"
+
+#include "ProcessLib/CoupledSolutionsForStaggeredScheme.h"
+
+namespace ProcessLib
+{
+namespace HT
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void StaggeredHTFEM<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assembleWithCoupledTerm(double const t,
+                            std::vector<double>& local_M_data,
+                            std::vector<double>& local_K_data,
+                            std::vector<double>& local_b_data,
+                            LocalCoupledSolutions const& coupled_xs)
+{
+    if (coupled_xs.process_id == 0)
+    {
+        assembleHeatTransportEquation(t, local_M_data, local_K_data,
+                                      local_b_data, coupled_xs);
+        return;
+    }
+
+    assembleHydraulicEquation(t, local_M_data, local_K_data, local_b_data,
+                              coupled_xs);
+}
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void StaggeredHTFEM<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assembleHydraulicEquation(double const t, std::vector<double>& local_M_data,
+                              std::vector<double>& local_K_data,
+                              std::vector<double>& local_b_data,
+                              LocalCoupledSolutions const& coupled_xs)
+{
+    auto const& local_p = coupled_xs.local_coupled_xs[1];
+    auto const local_matrix_size = local_p.size();
+    // This assertion is valid only if all nodal d.o.f. use the same shape
+    // matrices.
+    assert(local_matrix_size == ShapeFunction::NPOINTS);
+
+    auto const& local_T1 = coupled_xs.local_coupled_xs[0];
+    auto const& local_T0 = coupled_xs.local_coupled_xs0[0];
+    const double dt = coupled_xs.dt;
+
+    auto local_M = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<LocalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    auto local_b = MathLib::createZeroedVector<LocalVectorType>(
+        local_b_data, local_matrix_size);
+
+    SpatialPosition pos;
+    pos.setElementID(this->_element.getID());
+
+    auto const& material_properties = this->_material_properties;
+
+    auto const& b = material_properties.specific_body_force;
+
+    MaterialLib::Fluid::FluidProperty::ArrayType vars;
+
+    unsigned const n_integration_points =
+        this->_integration_method.getNumberOfPoints();
+
+    for (std::size_t ip(0); ip < n_integration_points; ip++)
```

Comment:
No need to extend an unsigned to size_t here.

Reply:
Changed accordingly.